psyneulink/core/llvm/codegen.py

```diff
@@ -293,9 +293,19 @@ def gen_composition_exec(ctx, composition, *, tags:frozenset):
 
         # Generate a while not 'end condition' loop
         builder.position_at_end(loop_condition)
+        is_finished_flags = {}
+        for idx, node in enumerate(composition.nodes):
+            try:
+                node_state = builder.gep(state, [ctx.int32_ty(0), ctx.int32_ty(0), ctx.int32_ty(idx)])
```

Comment:
this should use index into `nodes_states`

Reply:
not quite sure what you mean by this; can you give an example?

Reply:
use named memebr to access array of nodes' states instead of `ctx.int32_ty(0)`. it's already available on line 157, just move it up.

Reply:
my bad, wrong function.